subsys/net/lib/fota_download/src/fota_download.c

```diff
@@ -423,16 +423,37 @@ int fota_download(const char *host, const char *file,
 	const int *sec_tag_list, uint8_t sec_tag_count, uint8_t pdn_id, size_t fragment_size,
 	const enum dfu_target_image_type expected_type)
 {
-	uint32_t host_hash = 0;
-	uint32_t file_hash = 0;
-	int err = -1;
 	static int sec_tag_list_copy[CONFIG_FOTA_DOWNLOAD_SEC_TAG_LIST_SIZE_MAX];
-
 	struct download_client_cfg config = {
 		.pdn_id = pdn_id,
 		.frag_size_override = fragment_size,
 	};
```

Comment:
If we have the final goal of deprecating fota_download and eventually moving over to `fota_v2` API, I think we should have `if not set`, set `pdn+fragsize`, and have these be the same. It would give the least API change if we eventually are able to deprecate `fota_download`. However this is just an opinion and not something that really needs to be considered.

```C
config->pdn_id = pdn_id;
config->frag_size_override = fragment_size;
```

Reply:
The new API can replace below APIs
.fota_download()
.fota_download_any()
.fota_downlaod_start()
.fota_download_start_with_image_type()